tests/unit/package_managers/test_pip.py

```diff
@@ -3802,3 +3872,103 @@ def convert_to_component_list(packages: list[dict]) -> list[Component]:
     if n_pip_packages >= 2:
         mock_resolve_pip.assert_any_call(source_dir.join_within_root("foo"), output_dir, None, [])
         mock_replace_requirements.assert_any_call("/package_b/requirements.txt")
+
+
+@pytest.mark.parametrize(
+    "dependency, expected_purl",
+    [
+        (
+            {
+                "name": "pypi_package",
+                "version": "1.0.0",
+                "type": "pip",
+                "dev": False,
+                "kind": "pypi",
+            },
+            "pkg:pypi/pypi-package@1.0.0",
+        ),
+        (
+            {
+                "name": "git_dependency",
+                "version": f"git+https://github.com/my-org/git_dependency@{'a'*40}",
+                "type": "pip",
+                "dev": False,
+                "kind": "vcs",
+            },
+            f"pkg:pypi/git-dependency?vcs_url=git%2Bhttps://github.com/my-org/git_dependency%40{'a'*40}",
+        ),
+        (
+            {
+                "name": "Git_dependency",
+                "version": f"git+file:///github.com/my-org/git_dependency@{'a'*40}",
+                "type": "pip",
+                "dev": False,
+                "kind": "vcs",
+            },
+            f"pkg:pypi/git-dependency?vcs_url=git%2Bfile:///github.com/my-org/git_dependency%40{'a'*40}",
+        ),
+        (
+            {
+                "name": "git_dependency",
+                "version": f"git+ssh://git@github.com/my-org/git_dependency@{'a'*40}",
+                "type": "pip",
+                "dev": False,
+                "kind": "vcs",
+            },
+            f"pkg:pypi/git-dependency?vcs_url=git%2Bssh://git%40github.com/my-org/git_dependency%40{'a'*40}",
+        ),
+        (
+            {
+                "name": "git_dependency",
+                "version": f"git+https://github.com/my-org/git_dependency@{'a'*40}",
+                "type": "pip",
+                "dev": False,
+                "kind": "vcs",
+            },
+            f"pkg:pypi/git-dependency?vcs_url=git%2Bhttps://github.com/my-org/git_dependency%40{'a'*40}",
+        ),
+        (
+            {
+                "name": "https_dependency",
+                "version": f"https://github.com/my-org/https_dependency/{'a'*40}/file.tar.gz#egg=appr&cachito_hash=sha256:de526c1",
```

Comment:
Just FYI, the `#egg` must normally match the name of the dependency (but doesn't matter much here)